### PR TITLE
ci: Adopt ci changes from bitcoin

### PR DESCRIPTION
**Pull request by div72**
This PR aims to adopt the bitcoins CI scripts. This adds a new native macOS gui build and the ability to continue a build if the depends/build steps take too long. Also updates native_biplist, native_cctools, curl and the macOS SDK used by depends.

Enabling Qt Charts on macOS cross compilation fails with:
`make[3]: *** No rule to make target '/xxxxxxx/depends/work/build/x86_64-apple-darwin/qt59/5.9.6-7xxxxxxxx/qtbase/lib/Qt5Widgets.', needed by '../../lib/Qt5Charts.'.  Stop.`

~~While I feel like this is an improvement over the current system, I've pretty much hit a road-block in terms of improving this PR.~~

### CODE REVIEW

File: ci/test/00_setup_env_native.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# Copyright (c) 2019-2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+export LC_ALL=C.UTF-8
+
+export CONTAINER_NAME=ci_native
+export DOCKER_NAME_TAG=ubuntu:20.04
+export PACKAGES="libqt5gui5 libqt5core5a qtbase5-dev libqt5dbus5 qttools5-dev qttools5-dev-tools libqt5charts5-dev libssl-dev libevent-dev bsdmainutils libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-iostreams-dev libboost-program-options-dev libboost-test-dev libboost-thread-dev libdb5.3++-dev libminiupnpc-dev libqrencode-dev libzip-dev zlib1g zlib1g-dev libcurl4 libcurl4-openssl-dev"
+export RUN_UNIT_TESTS=false
```
Comment (jamescowens) [inline ci/test/00_setup_env_native.sh]: This is not exactly the same as our dependency list in .travis.yml. Also, shouldn't RUN_UNIT_TESTS be true here (and maybe false elsewhere?

File: ci/test/00_setup_env_win32.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+export LC_ALL=C.UTF-8
+
+export CONTAINER_NAME=ci_win32
+export DOCKER_NAME_TAG=ubuntu:18.04  # Check that bionic can cross-compile to win32 (bionic is used in the gitian build as well)
+export HOST=i686-w64-mingw32
+export PACKAGES="python3 nsis g++-mingw-w64-i686 wine-binfmt wine32"
+export RUN_UNIT_TESTS=false
+export RUN_FUNCTIONAL_TESTS=false
```
Comment (jamescowens) [inline ci/test/00_setup_env_win32.sh]: We currently don't include win-binfmt or wine32/64, and I would not think it is necessary unless you are going to run the unit tests for the windows executable under WINE.

File: ci/test/00_setup_env_win64.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+export LC_ALL=C.UTF-8
+
+export CONTAINER_NAME=ci_win64
+export DOCKER_NAME_TAG=ubuntu:18.04  # Check that bionic can cross-compile to win64 (bionic is used in the gitian build as well)
+export HOST=x86_64-w64-mingw32
+export PACKAGES="python3 nsis g++-mingw-w64-x86-64 wine-binfmt wine64"
+export RUN_UNIT_TESTS=true
```
Comment (jamescowens) [inline ci/test/00_setup_env_win64.sh]: Same comment here. Why would we enable unit tests here and not on win32?

**Comment by jamescowens** [comment]
Some of those failures look like trivial errors in the test pass qualifications. We probably would be getting the same failures in current travis, but we are running the tests on the native compile only.

File: ci/test/00_setup_env_native.sh (hunk 1 above)
Comment (div72) [reply ci/test/00_setup_env_native.sh]: It was originally turned on, I turned it off after getting the `test/fs_tests.cpp` error in the x86_64 build.

File: ci/test/00_setup_env_win32.sh (hunk 2 above)
Comment (div72) [reply ci/test/00_setup_env_win32.sh]: Also turned this off after `test/fs_tests.cpp` error. Forgot to remove the dependencies, will do.

File: ci/test/00_setup_env_win64.sh (hunk 3 above)
Comment (div72) [reply ci/test/00_setup_env_win64.sh]: The tests actually pass in win64, I can still disable it if you want?

**Comment by jamescowens** [comment]
We should be able to get the unit tests to pass on the native build. They pass on the native build in the current Travis environment.

**Comment by jamescowens** [comment]
@cyrossignol please take a look at the unit test failures above. I would like to get this PR to work and merged. It is a vast improvement over what we are doing currently.

File: ci/test/00_setup_env_win64.sh (hunk 3 above)
Comment (jamescowens) [reply ci/test/00_setup_env_win64.sh]: We should actually try and get the tests to pass on all platforms. I also want to figure out why the qtcharts is giving a missing rule in make. Is there a missing dependency?

**Comment by jamescowens** [comment]
This one is going to get merged before the corresponding Github Workflow one (#1746).

**Review by jamescowens — APPROVED** [review]

**Pull request merged**